akka-bench-jmh/src/main/scala/akka/actor/ActorThoughputBenchmark.scala

```diff
@@ -0,0 +1,223 @@
+/**
+ * Copyright (C) 2014-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.actor
+
+import akka.pattern.ask
+import akka.pattern.pipe
+import akka.testkit.TestProbe
+import com.typesafe.config.ConfigFactory
+import org.openjdk.jmh.annotations._
+import scala.concurrent.duration._
+import java.util.concurrent.TimeUnit
+import scala.concurrent.Await
+import scala.annotation.tailrec
+import java.util.concurrent.CountDownLatch
+import akka.actor.SupervisorSpec.PingPongActor
+import akka.util.Timeout
+import scala.concurrent.Future
+import scala.concurrent.forkjoin.ThreadLocalRandom
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Array(Mode.Throughput))
+@Fork(1)
+@Threads(1)
+@Warmup(iterations = 10, time = 5, timeUnit = TimeUnit.SECONDS, batchSize = 1)
+@Measurement(iterations = 10, time = 15, timeUnit = TimeUnit.SECONDS, batchSize = 1)
+class ActorThroughputBenchmark {
```

Comment:
`Will certainly give it a spin tomorrow morning. The behavior of the mailbox is quite interesting as well. If this is the case, this could actually tip off the benchmark a bit. AFAIK, the behavior of the lock free queues such as the one used in the Mailbox changes quite a bit depending on their average occupancy. Having it mostly at size 1 could skew results.